lib/qam.pm

```diff
@@ -216,5 +217,12 @@ sub get_packagebins_in_modules {
     return map { $_->{name} => $_ } @arr;
 }
 
+sub repo_is_not_active {
+    my $repo = $_[0];
+    $repo =~ m".+Maintenance\:\/(\d+)";
+    my $id     = $1;
+    my $status = query_smelt("{incidents(incidentId: $id){edges{node{status {name}}}}}");
+    return 1 && record_info('Not added', "$id have been released or declined") if $status =~ /done/;
```

Comment:
Updates that have been declined remain active. For example querying `{  incidents(incidentId:16433 ){edges {node {status {name} }}}}` returns `"active"`. This tests only if they have been released.

Since you are looking for the value of a string field, it's better that you either parse the JSON into a hash and extract the value from there or if you don't, at least match for `"status":{"name": "done"}`. The way you do it now will break silently, if a value like `not_done` is added in the future.

Reply:
Yes `declined` is still active, but that is not breaking tests because declined is not revoked, the repo still does exist.
I will check this and test also declined, revoked updates.

I agree with extracting specific value, but I don't think it's needed here. Interesting speculation, but there is no reason `not_done` would be added. Incident is either `active` or `done` AFAIK.

Reply:
changed to `status":{"name":"done"`
`declined` updates are mostly reopened again with additional changes, so still active.
I think it would be good to don't add also that but it was never a problem.